### PR TITLE
Support encoding with fastavro, fix minor doc bug

### PR DESCRIPTION
**Pull request by marcintustin**
This PR adds the ability to encode with fastavro, if it is available. Also, fixes a minor misprecision in the docstring for `encode_record_with_schema_id`.

### CODE REVIEW

**Comment by marcintustin** [comment]
[clabot:check]

**Comment by marcintustin** [comment]
The failure here appears to be with one of the workers timing out. I can't force a re-run, I'm afraid.

**Comment by rnpridgeon** [comment]
Thanks @marcintustin! I'll take a look tomorrow, in the meantime I have forced a restart of the failed worker.

File: confluent_kafka/avro/serializer/message_serializer.py [hunk 1]
```diff
@@ -101,34 +108,35 @@ def encode_record_with_schema(self, topic, schema, record, is_key=False):
...
-        # use slow avro
         if schema_id not in self.id_to_writers:
             # get the writer + schema
-
             try:
                 schema = self.registry_client.get_by_id(schema_id)
                 if not schema:
                     raise serialize_err("Schema does not exist")
-                self.id_to_writers[schema_id] = avro.io.DatumWriter(schema)
+                self.id_to_writers[schema_id] = self.make_datum_writer(schema)
```
Comment (rnpridgeon) [inline confluent_kafka/avro/serializer/message_serializer.py]: I would move this logic back into `_encode_record_with_schema_id` and eliminate the redundant `_slow_encode_record_with_schema_id` and `_fast_encode_record_with_schema_id` methods. If we provide the same interface for both writers there is no longer a need to maintain two code paths which is a bit more DRY.

**Comment by rnpridgeon** [comment]
Thanks for the contribution @marcintustin , I went ahead and made the requested changes in #492 to try and get this in for the 1.0 release which enters code freeze at EOD.

**Pull request closed without merging**